POLMED/src/components/nav/index.tsx

```diff
@@ -1,40 +1,53 @@
+import { useEffect, useState } from "react";
 import { Link, useLocation } from "react-router-dom";
-import NavItem from "./navItems";
-import { navItems, NavItemType } from "./navItems/navItems";
 import logo from "@/img/logo/logo.png";
 import whiteLogo from "@/img/logo/logo_white.png";
+import NavUl from "./components/navUl";
 
 const Nav = () => {
   const location = useLocation();
   const { pathname: path } = location;
 
+  const [windowSize, setWindowSize] = useState([window.innerWidth, window.innerHeight]);
+
+  useEffect(() => {
+    const handleWindowResize = () => {
+      setWindowSize([window.innerWidth, window.innerHeight]);
+      if (window.innerWidth > 1024) {
+        const navUl = document.querySelector(".nav-ul");
+        (navUl as HTMLElement).style.setProperty("transform", "translateX(0)");
+      }
+    };
+
+    window.addEventListener("resize", handleWindowResize);
+
+    return () => {
+      window.removeEventListener("resize", handleWindowResize);
+    };
+  });
+
+  const toggleMenu = () => {
+    const navUl = document.querySelector(".nav-ul");
+    console.log(navUl);
+    (navUl as HTMLElement).style.setProperty("transform", "translateX(-100%)");
+  };
+
   return (
-    <nav className={`${path === "/" ? "nav-hp" : "nav"}`}>
+    <nav className={`nav ${path === "/" ? "nav-hp" : "nav-other"}`}>
       <Link to="/">
         <img className="nav-logo" src={path === "/" ? whiteLogo : logo} alt="logo" />
       </Link>
-      <ul className="nav-ul">
-        {path === "/"
-          ? navItems.map((item: NavItemType) => (
-              <NavItem
-                key={item.content}
-                content={item.content}
-                url={item.url}
-                homePage={true}
-                active={path === item.url}
-              />
-            ))
-          : navItems.map((item: NavItemType) => (
-              <NavItem
-                key={item.content}
-                content={item.content}
-                url={item.url}
-                homePage={false}
-                active={path === item.url}
-              />
-            ))}
-        <button className={`nav-btn ${path === "/" && "nav-btn-hp"}`}>Wyloguj</button>
-      </ul>
+      {windowSize[0] <= 1024 && (
+        <i
+          onClick={toggleMenu}
+          className={`${"nav-bars-icon"} ${path === "/" ? "nav-bars-icon-white" : ""}`}
```

Comment:
Proponuje zrobić jakąś zmienną globalną, która będzie nas informować o tym czy jesteśmy na stronie głównej.

Reply:
próbowałem to zrobić, ale doszedłem do wniosku, że to będzie trochę bez sensu - mianowicie tworząc context, który mówi nam na jakiej ścieżce się znajdujemy lub czy jesteśmy na stronie głównej i tak stworzymy go używając useLocation, bo to jest najprostszy sposób na zdobycie dokładnej lokalizacji i w zasadzie jest to gotowy hook do użycia. Używając go dla jakiejś zmiennej lokalnej musielibyśmy nasłuchiwać czy przypadkiem ta lokacja się nie zmienia, bo jeśli przypiszemy zmienną do tego co zwróci nam useLocation i sprawimy, że będzie dostępna globalnie to globalnie dostępne będzie do co nam useLocation zwróci za 1 razem, a zmieniając ścieżkę będziemy mieli problem, bo skąd useLocation ma wiedzieć o tym, że musi się wywołać drugi raz? To my w jakiś sposób musielibyśmy wiedzieć, że trzeba jeszcze raz użyć useLocation, bo ścieżka się zmieniła, a tym sposobem dochodzimy do punktu wyjścia, czyli tego jak używaliśmy useLocation dotychczas - używamy go na każdym renderze danego komponentu, w którym potrzebujemy jej użyć. Reasumując, wydaję mi się, że w tym przypadku użycie jakiegokolwiek globalnego contextu będzie mijało się z celem, a kod, który wówczas napiszemy będzie wyglądał trochę tak jakbyśmy nie napisali, że 2 + 2 = 4, tylko zamiast tego coś takiego: 2 + 2 = (2^(-1))*4 + (2^(-1))*4 = 4

Jeśli masz jakieś inne rozwiązanie, które można byłoby tu zaimplementować to daj znać i może uda się coś sensownego napisać

Reply:
Spróbuję jeszcze zastosować jedno rozwiązanie, może się uda coś wskórać

Reply:
coś wykminiłem, zobaczysz czy spoko